pywbem/cim_operations.py

```diff
@@ -1328,112 +1326,135 @@ def _configure_detail_level(cls, detail_level):
+
+        The specified logger is always a single pywbem logger; the simple
+        logger name 'all' has already been resolved by the caller into
+        multiple calls to this method.
 
-        If connection parameter is boolean and True, sets class level attribute
-        so all subsequent WBEMConnection objects are activated and sets detail
-        level in class parameter.
+        The 'handler' parameter controls logger configuration:
+        * If None, nothing is done.
+        * If a Handler object, the specified logger gets its handlers replaced
+          with the new handler, and logging level DEBUG is set.
+
+        The 'connection' paraneter controls activation and setting of the
+        detail level:
+        * If None, nothing is done.
+        * If bool=True, log activation and log detail information is stored for
+          use by future connections in class variables of WBEMConnection.
+        * If a WBEMConnection object, logging is activated and detail level is
+          set immediately for that connection.
         """
 
-        # We allow only a single handler for any logger so must remove any
-        # existing handler before adding
-        # TODO why limit of single handler???
-        if handler:
+        if handler is not None:
+            assert isinstance(handler, logging.Handler)
+
+            # Replace existing handlers of the specified logger (e.g. from
+            # previous calls) by the specified handler
+
             logger = logging.getLogger(logger_name)
             for hdlr in logger.handlers:
                 logger.removeHandler(hdlr)
             logger.addHandler(handler)
-            logger.setLevel(logging.DEBUG)
 
-        # TODO should force cimple name above.
-        logger_simple_name = logger_name.split('.')[1]
+            logger.setLevel(logging.DEBUG)
 
-        if connection:
+        if connection is not None:
             if isinstance(connection, bool):
+                if not connection:
+                    raise ValueError("Invalid boolean connection: %r; must "
+                                     "be True" % connection)
+
+                # Store the activation and detail level information for future
+                # connections. This information is used in the init method
+                # of WBEMConnection to activate logging at that point.
+
                 cls._activate_logging = True
+                cls._log_detail_levels[simple_name] = detail_level
+
             else:
                 assert isinstance(connection, WBEMConnection)
-                local_dict = cls._log_config_dict.copy()
-                local_dict[logger_simple_name] = detail_level
 
-                # Insert recorder into list if not already there.
-                # pylint: disable=protected-access
-                recorder_found = False
+                # Activate logging for this existing connection by ensuring
+                # the connection has a log recorder
```

Comment:
I think I found an issue here that I fixed yesterday in the pr for issue #1167.  Will look later.

Reply:
**Discussion point:** We should make sure that the two PRs can come together.